e2e/internal/e2e/imageverify.go

```diff
@@ -20,84 +20,41 @@ import (
 
 // ImageVerify checks for an image integrity.
 func (env TestEnv) ImageVerify(t *testing.T, imagePath string) {
-	tt := []struct {
-		name string
-		argv []string
-		exit int
-	}{
-		{
-			name: "False",
-			argv: []string{imagePath, "false"},
-			exit: 1,
-		},
-		{
-			name: "RunScript",
-			argv: []string{imagePath, "test", "-f", "/.singularity.d/runscript"},
-			exit: 0,
-		},
-		{
-			name: "OneBase",
-			argv: []string{imagePath, "test", "-f", "/.singularity.d/env/01-base.sh"},
-			exit: 0,
-		},
-		{
-			name: "ActionsShell",
-			argv: []string{imagePath, "test", "-f", "/.singularity.d/actions/shell"},
-			exit: 0,
-		},
-		{
-			name: "ActionsExec",
-			argv: []string{imagePath, "test", "-f", "/.singularity.d/actions/exec"},
-			exit: 0,
-		},
-		{
-			name: "ActionsRun",
-			argv: []string{imagePath, "test", "-f", "/.singularity.d/actions/run"},
-			exit: 0,
-		},
-		{
-			name: "Environment",
-			argv: []string{imagePath, "test", "-L", "/environment"},
-			exit: 0,
-		},
-		{
-			name: "Singularity",
-			argv: []string{imagePath, "test", "-L", "/singularity"},
-			exit: 0,
-		},
-	}
-
-	for _, tc := range tt {
-		env.RunSingularity(
-			t,
-			AsSubtest(tc.name),
-			WithProfile(UserProfile),
-			WithCommand("exec"),
-			WithArgs(tc.argv...),
-			ExpectExit(tc.exit),
-		)
-	}
+	env.RunSingularity(
+		t,
+		AsSubtest("BasicIntegrityTests"),
+		WithProfile(UserProfile),
+		WithCommand("exec"),
+		WithArgs(imagePath, "/bin/sh", "-c",
+			"test -f /.singularity.d/runscript -a "+
+				"-f /.singularity.d/env/01-base.sh -a "+
+				"-f /.singularity.d/actions/shell -a "+
+				"-f /.singularity.d/actions/exec -a "+
+				"-f /.singularity.d/actions/run -a "+
+				"-L /environment -a "+
+				"-L /singularity"),
+		ExpectExit(0),
+	)
 
 	tests := []struct {
-		name      string
+		// name      string
 		jsonPath  []string
 		expectOut string
 	}{
 		{
-			name:      "LabelCheckType",
+			// name:      "LabelCheckType",
 			jsonPath:  []string{"type"},
 			expectOut: "container",
 		},
 		{
-			name:      "LabelCheckSchemaVersion",
+			// name:      "LabelCheckSchemaVersion",
 			jsonPath:  []string{"data", "attributes", "labels", "org.label-schema.schema-version"},
 			expectOut: "1.0",
 		},
 	}
 
-	// Verify the label partition
-	for _, tt := range tests {
-		verifyOutput := func(t *testing.T, r *SingularityCmdResult) {
+	verifyOutput := func(t *testing.T, r *SingularityCmdResult) {
+		for _, tt := range tests {
 			jsonOut, err := jsonparser.GetString(r.Stdout, tt.jsonPath...)
```

Comment:
We should be able to bring back the named subtests here, by wrapping the repeated statements with...

`t.Run(tt.name, func(t *testing.T) {`

There are only 2 now ... so it might not seem worth it, but really we should add a lot more checks on the labels at some point.